### PR TITLE
checker: fix sum type operator check

### LINKED ISSUE #6814: Sum type infix operations should be disallowed
**V version:** custom branch
**OS:** Windows 10 WSL2 Ubuntu 20.04

**What did you do?**
```v
type Abc = int | string

fn main() {
	x := Abc(t)
	_ := x + Abc(5)
}
```

**What did you expect to see?**
an v compiler error
 
**What did you see instead?**
a c error

### PR DESCRIPTION
**Pull request by danieldaeschle**
This fixes #6814

### CODE REVIEW

**Comment by danieldaeschle** [comment]
Fixes #6749

File: vlib/v/checker/tests/sum_type_infix_err.vv [hunk 1]
```diff
@@ -0,0 +1,7 @@
+type Abc = int | string
+
+fn main() {
+	x := Abc(0)
+	_ := x + Abc(5)
+	_ := 'test' + x
```
Comment (ntrel) [inline vlib/v/checker/tests/sum_type_infix_err.vv:6]: It looks like this pull may forbid infix operators on a pointer to a sum type, or an optional sum type.
Could you add a line e.g. `_ = unsafe{&x + 5}` to ensure no error message for that please?

**Review by ntrel — CHANGES_REQUESTED** [review]

File: vlib/v/checker/tests/sum_type_infix_err.vv (hunk 1 above)
Comment (danieldaeschle) [reply vlib/v/checker/tests/sum_type_infix_err.vv:6]: Ok, it fails. Do you have an idea how we could make it work?

Comment (danieldaeschle) [reply vlib/v/checker/tests/sum_type_infix_err.vv:6]: is_ptr check should help

**Comment by ntrel** [comment]
Thanks for adding the test. I made a new method to get the kind from a Type that respects pointers and optionals as well, and used that with your branch:
https://github.com/ntrel/V/commits/sum-kind
Maybe cherry-pick that commit?
I think this method would be useful in a lot of places that ignore pointers etc.

**Comment by danieldaeschle** [comment]
yeah, you can push it on my branch as well if you want.

**Comment by danieldaeschle** [comment]
@ntrel is it ok?

**Comment by danieldaeschle** [comment]
@ntrel your commit doesn't work on all scenarios.

**Comment by ntrel** [comment]
How do you mean?

**Comment by Delta456** [comment]
I think its because the CI is failing.

**Comment by danieldaeschle** [comment]
it broke something which worked before

**Comment by danieldaeschle** [comment]
No, sorry. i just had to fix the error messages from the other PR.

**Comment by danieldaeschle** [comment]
@ntrel can you resolve your change request?

**Comment by danieldaeschle** [comment]
@spytheman can you merge this?

**Review by ntrel — APPROVED** [review]

**Pull request merged**